tensorflow/core/grappler/optimizers/auto_mixed_precision.h

```diff
@@ -17,17 +17,17 @@ limitations under the License.
 #define TENSORFLOW_CORE_GRAPPLER_OPTIMIZERS_AUTO_MIXED_PRECISION_H_
 
 #include "tensorflow/core/grappler/optimizers/graph_optimizer.h"
+#include "tensorflow/core/protobuf/device_properties.pb.h" 
 #include "tensorflow/core/protobuf/rewriter_config.pb.h"
 
 namespace tensorflow {
 namespace grappler {
 
-enum class AutoMixedPrecisionMode { CUDA, MKL };
-
-//Getting FP16 supported devices for ROCm
 #if TENSORFLOW_USE_ROCM
-bool HasEnhancedFP16ComputeSupport(std::pair<int, int> gpu_arch);
-#endif
+    bool GetFastFP16Support(const DeviceProperties& props); 
+#endif 
```

Comment:
Put this after the enum decl for `AutoMixedPrecisionMode` to avoid the unnecessary diff....will be helpful when you file the upstream PR

Also make sure you have run clang-format

Reply:
actually we may not need this routine in the header file at all...see comments below